website/static/js/setPassword.js

```diff
@@ -0,0 +1,267 @@
+            required: true
+        });
+
+        validatedFields.oldPassword = self.oldPassword;
+    }
+
+    // only include the following fields if the user is
+    // signing up for the first time
+    if (passwordViewType === 'signup') {
+        self.fullName = ko.observable('').extend({
+            required: true,
+            minLength: 3
+        });
+        self.email1 = ko.observable('').extend({
+            required: true,
+            email: true
+        });
+        self.email2 = ko.observable('').extend({
+            required: true,
+            email: true,
+            validation: {
+                validator: function(val, other) {
+                    return String(val).toLowerCase() === String(other).toLowerCase();
+                },
+                'message': 'Email addresses must match.',
+                params: self.email1
+            }
+        });
+
+        self.password.extend({
+            validation: {
+                validator: function(val, other) {
+                    if (String(val).toLowerCase() === String(other).toLowerCase()) {
+                        self.typedPassword(' ');
+                        return false;
+                    } else {
+                        return true;
+                    }
+                },
+                'message': 'Your password cannot be the same as your username.',
+                params: self.email1
+            }
+        });
+
+        validatedFields.fullName = self.fullName;
+        validatedFields.email1 = self.email1;
+        validatedFields.email2 = self.email2;
+
+    }
+
+    // pick up the email from contextVars if we can't get it from first typing it in
+    if (window.contextVars.username) {
+        self.email1 = ko.observable(window.contextVars.username);
+    }
+
+    // If we have gotten an email to compare to at this point, also validate against that
+    if (self.email1) {
+        self.password.extend({
+            validation: {
+                validator: function(val, other) {
+                    if (String(val).toLowerCase() === String(other).toLowerCase()) {
+                        self.typedPassword(' ');
+                        return false;
+                    } else {
+                        return true;
+                    }
+                },
+                'message': 'Your password cannot be the same as your username.',
+                params: self.email1
+            }
+        });
+    }
+
+    // Collect validated fields
+    self.validatedFields = ko.validatedObservable($.extend({}, validatedFields));
+
+    self.submitted = ko.observable(false);
+
+    self.flashMessage = ko.observable('');
+    self.flashMessageClass = ko.observable('');
+    self.flashTimeout = null;
+
+    self.trim = function(observable) {
+        observable($.trim(observable()));
+    };
+
+    /** Change the flashed message. */
+    self.changeMessage = function(message, messageClass, text, css, timeout, timeoutClock) {
```

Comment:
This can be simplified; why pass both `message` and `text`? You should just be able to pass one of these. Passing a string message as the first argument.

```js
self.changeMessage('Password reset successful!', ...)
```

You already have an observable for storing the flashed message.

```js
self.changeMessage(message, ...) {
     self.flashMessage(message);
}